packages/nightingale-linegraph-track/src/nightingale-linegraph-track.js

```diff
@@ -0,0 +1,149 @@
+    this.svg = select(this)
+      .append("svg")
+      .attr("width", this.width)
+      .attr("height", this._height);
+    this.trackHighlighter.appendHighlightTo(this.svg);
+
+    const range = [];
+    this._data.map((d) => range.push(...d.range));
+    this.minRange = min(range);
+    this.maxRange = max(range);
+
+    // Create the visualisation here
+    this.refresh();
+  }
+
+  refresh() {
+    if (!this.svg) return;
+    this.svg.selectAll("g.chart-group").remove();
+    this.chartGroup = this.svg.append("g").attr("class", "chart-group");
+    this._initYScale();
+    this._checkSeqBaseWidth();
+
+    this.chart = this.chartGroup
+      .selectAll(".chart")
+      .data(this._data)
+      .enter()
+      .append("g")
+      .attr("class", "chart");
+
+    this.chart
+      .append("path")
+      .attr("class", "graph")
+      .attr("id", (d) => d.name)
+      .attr("d", (d) => {
+        this.drawLine(d);
+        d.colour = d.colour || interpolateRainbow(Math.random()); // eslint-disable-line no-param-reassign
+        return this.line(d.values);
+      })
+      .attr("fill", "none")
+      .attr("stroke", (d) => d.colour)
+      .attr("transform", "translate(0,0)")
+      .on("mouseover", (d) => {
+        if (this.isSeqBaseVisible) {
+          const coords = mouse(this);
+          this._createEvent(coords, d, "mouseover");
+        }
+      })
+      .on("click", (d) => {
+        if (this.isSeqBaseVisible) {
+          const coords = mouse(this);
+          this._createEvent(coords, d, "click");
+        }
+      });
+
+    this._updateHighlight();
+  }
+
+  _initYScale() {
+    this._yScale
+      .domain([this.minRange, this.maxRange])
+      .range([this._height, 0]);
+  }
+
+  drawLine(d) {
+    this._curve = d.lineCurve || "curveLinear";
+
+    this.line = line()
+      .x(
+        (d) =>
+          this.getXFromSeqPosition(d.position + 1) -
+          this.getSingleBaseWidth() / 2
+      )
+      .y((d) => this._yScale(d.value))
+      .curve(d3[this._curve]);
+  }
+
+  _createEvent(coords, data, type) {
+    this.addPointer(coords);
+    const seqPosition = Math.floor(
+      this.xScale.invert(coords[0] - this.getSingleBaseWidth() / 2)
+    );
+    const [value] = data.values.filter((v) => v.position === seqPosition);
+    const detail = {
+      eventtype: type,
+      // coords: coords,
+      feature: value,
+      highlight: `${seqPosition}:${seqPosition}`,
+    };
+    this.dispatchEvent(
+      new CustomEvent("change", {
+        detail,
+        bubbles: true,
+        cancelable: true,
+      })
+    );
+  }
+
+  _checkSeqBaseWidth() {
+    const xratio = 0.8;
+    const tempNode = this.svg.append("text").attr("class", "base").text("T");
+    const chWidth = tempNode.node().getBBox().width * xratio;
+    tempNode.remove();
+    const ftWidth = this.getSingleBaseWidth();
+    this.isSeqBaseVisible = ftWidth - chWidth > 1;
+  }
+
+  addPointer(coords) {
+    this.chart.selectAll("circle.pointer").remove();
```

Comment:
Actually wondering if it would be better to have small semi-transparent circles always visible at the end of lines, and have the event be attached to these rather than the lines?

Reply:
Could you explain it? Meaning having circles at the data points and attaching the events to the circles?

Reply:
I was saying that because I noticed a performance issue when mousing over the lines. I think this is happening because the event keeps getting triggered over and over again and it causes a backlog?
Actually a better approach might be something like [this](https://bl.ocks.org/larsenmtl/e3b8b7c2ca4787f77d78f58d41c3da91)?

Reply:
Thank you. It looks very cool. I will try to have something like this.